DQM/Physics/src/FSQDQM.cc

```diff
@@ -0,0 +1,331 @@
+   h_leadingtrkpt_ptsum_towards =bei.bookProfile("h_leadingtrkpt_ptsum_towards","h_leadingtrkpt_ptsum_towards",50,0,50,0,30," ");
+   h_leadingtrkpt_ptsum_transverse =bei.bookProfile("h_leadingtrkpt_ptsum_transverse","h_leadingtrkpt_ptsum_transverse",50,0,50,0,30," ");
+     
+ }
+
+
+
+
+
+
+ // ------------ method called for each event  ------------
+ void
+ FSQDQM::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+ {
+   using namespace edm;
+   using namespace std;
+   using namespace reco;
+
+    runNumber_   = iEvent.id().run();
+    eventNumber_ = iEvent.id().event();
+    lumiNumber_  = iEvent.id().luminosityBlock();
+    bxNumber_ = iEvent.bunchCrossing();
+
+    edm::Handle<edm::View<reco::Vertex> > privtxs;
+    //    if(!iEvent.getByToken(pvs_, privtxs)) return;
+    iEvent.getByToken(pvs_, privtxs);
+    double bestvz=-999.9, bestvx=-999.9, bestvy=-999.9;
+    double bestvzError=-999.9, bestvxError=-999.9, bestvyError=-999.9;
+    if(privtxs.isValid()){
+      const reco::Vertex& pvtx=privtxs->front();
+      NPV->Fill(privtxs->size());
+      if(privtxs->begin() !=privtxs->end() && !(pvtx.isFake()) && pvtx.position().Rho() <= 2. && fabs(pvtx.position().z()) <= 24){
+	bestvz = pvtx.z(); 
+	bestvx = pvtx.x(); 
+	bestvy = pvtx.y();
+	bestvzError = pvtx.zError(); 
+	bestvxError = pvtx.xError(); 
+	bestvyError = pvtx.yError();
+	PV_chi2->Fill(pvtx.normalizedChi2());
+	PV_d0->Fill(sqrt(pvtx.x() * pvtx.x() + pvtx.y() * pvtx.y()));
+	PV_numTrks->Fill(pvtx.tracksSize());
+	double vertex_sumTrks = 0.0;
+	for(reco::Vertex::trackRef_iterator iTrack= pvtx.tracks_begin(); iTrack != pvtx.tracks_end(); iTrack++)
+	  {
+	    vertex_sumTrks += (*iTrack)->pt();
+	  }
+	PV_sumTrks->Fill(vertex_sumTrks);
+      }
+    }
+    
+    std::vector<Jet> recoPFJets;
+	 recoPFJets.clear();
+	 int  nPFCHSJet=0;
+	 edm::Handle<PFJetCollection> pfjetchscoll;
+	 //	 if(!iEvent.getByToken(tok_pfjet_, pfjetchscoll)) return;
+	 iEvent.getByToken(tok_pfjet_, pfjetchscoll);
+	 if(pfjetchscoll.isValid()){
+	 const reco::PFJetCollection *pfchsjets = pfjetchscoll.product();
+	 reco::PFJetCollection::const_iterator pfjetchsclus = pfchsjets->begin();
+	 for(pfjetchsclus = pfchsjets->begin(); pfjetchsclus!= pfchsjets->end() ; ++pfjetchsclus){
+	   PFJetpt->Fill( pfjetchsclus->pt());
+	   PFJeteta->Fill( pfjetchsclus->eta());
+	   PFJetphi->Fill( pfjetchsclus->phi());
+	   PFJetRapidity->Fill( pfjetchsclus->rapidity());
+	   nPFCHSJet++;
+	 }
+	 PFJetMulti->Fill( nPFCHSJet);
+	 }
+ 
+	 std::vector<Jet> recoCastorJets;
+         recoCastorJets.clear();
+	 edm::Handle<BasicJetCollection> castorJets;
+	 //	 if(!iEvent.getByToken(tok_castorjet_, castorJets)) return;
+	 iEvent.getByToken(tok_castorjet_, castorJets);
+	 if(castorJets.isValid()){
+	 for (unsigned ijet=0; ijet<castorJets->size();ijet++) {
+	   recoCastorJets.push_back((*castorJets)[ijet]);
+	 }
+	 for (unsigned ijet=0; ijet<recoCastorJets.size(); ijet++) {
+	   cout<<recoCastorJets[ijet].pt()<<endl;
```

Comment:
@anmehta @dmitrijus this is adding lots of printouts to the output, please eliminate it!